File: CC2 - Midterms (1Y 1S)/Calculator.java
/**
 * @author pao
 * BSCS 1B
 */
import java.util.Scanner;
public class Calculator {
    public static void main(String[] args) {
        Scanner sc = new Scanner (System.in);
        int num1, num2, result;
        String operator;
        System.out.println("---------------------------------------");
        System.out.println("             Java Calculator           ");
        System.out.println("---------------------------------------");
        
        System.out.print("Enter First Number                : ");
        num1 = sc.nextInt();
        System.out.print("Enter the Operator (+, -, *, /, %): ");
        operator = sc.next();
        System.out.print("Enter Second Number               : ");
        num2 = sc.nextInt();
        
        switch (operator){
            case "+":
                result = num1+num2;
                System.out.println("---------------------------------------");
                System.out.println("                           Result: " + result);
                System.out.println("---------------------------------------");
                break;
            case "-":
                result = num1 - num2;
                System.out.println("---------------------------------------");
                System.out.println("                           Result: " + result);
                System.out.println("---------------------------------------");
                break;
            case "*":
                result = num1 * num2;
                System.out.println("---------------------------------------");
                System.out.println("                           Result: " + result);
                System.out.println("---------------------------------------");
                break;
            case "/":
                if (num2 == 0){
                    System.out.println("-------------------------------------------");
                    System.out.println("Error: Division by Zero (0) is not allowed.");
                    System.out.println("-------------------------------------------");
                } else {
                    result = num1/num2;
                    System.out.println("---------------------------------------");
                    System.out.println("                           Result: " + result);
                    System.out.println("---------------------------------------");
                }              
                break;
            case "%":
                if (num2 == 0) {
                    System.out.println("-----------------------------------------");
                    System.out.println("Error: Modulo by Zero (0) is not allowed.");
                    System.out.println("-----------------------------------------");
                } else {
                    result = num1%num2;
                    System.out.println("---------------------------------------");
                    System.out.println("                           Result: " + result);
                    System.out.println("---------------------------------------");
                }
                break;
            default:
                System.out.println("Error: Invalid operator.");
                break;
        }
    }
}

File: Practice Codes/CC2 Java to Python/Midterms/Calculator.py
print("---------------------")
print("| Python Calculator |")
print("---------------------")

num1 = float(input("Enter first number        : "))
operator = input("Enter operator (+,-,*,/,%): ")
num2 = float(input("Enter second number       : "))

match operator:
    case "+":
        result = num1 + num2
        print("---------------------------")
        print("               Result: " + str(result))
        print("---------------------------")
    case "-":
        result = num1 - num2
        print("---------------------------")
        print("               Result: " + str(result))
        print("---------------------------")
    case "*":
        result = num1 * num2
        print("---------------------------")
        print("               Result: " + str(result))
        print("---------------------------")
    case "/":
        if num2 == 0:
            print("----------------------------------------------")
            print("| Error: Division by Zero (0) is not allowed |")
            print("----------------------------------------------")
        else:
            result = num1 / num2
            print("---------------------------")
            print("               Result: " + str(result))
            print("---------------------------")
    case "%":
        if num2 == 0:
            print("--------------------------------------------")
            print("| Error: Modulo by Zero (0) is not allowed |")
            print("--------------------------------------------")
        else:
            result = num1 % num2
            print("---------------------------")
            print("               Result: " + str(result))
            print("---------------------------")
    case _:
        print("Invalid operator. Try again")
